### PR TITLE
Refactor api and types 

### PR DESCRIPTION
**Pull request by olehmell**

### CODE REVIEW

File: packages/api/src/subsocial/basic.ts [hunk 1]
```diff
@@ -1,107 +1,215 @@
...
+  async findUnlistedPosts (ids: AnySpaceId[]) {
+    return this.findPosts({ ids, visibility: 'onlyUnlisted' })
+  }
+
+  /** Find and load posts with their extension and owner's profile (if defined). */
+  async findPostsWithSomeDetails (filter: FindPostsWithDetailsQuery): Promise<RawPostWithSomeDetails[]> {
+    const posts = await this.findPosts(filter)
+    return loadAndSetPostRelatedStructs(posts, this.structFinders, filter)
+  }
+
```
Comment (tusharojha) [inline packages/api/src/subsocial/basic.ts:96]: withSomeDetails is not easy to understand. 
Let's rename it to `findPostWithOwnerDetails`

File: packages/api/src/subsocial/flatteners/index.ts [hunk 2]
```diff
@@ -62,19 +62,19 @@ function flattenSpaceOrPostStruct (struct: SpaceOrPostStruct): FlatSpaceOrPost {
   }
 }
 
-export const flattenPermisions = (permissions?: SpacePermissions) => {
-  const flatPermissions: FlatSpacePermissions = {}
+export const flattenPermisions = (permissions?: BlockchainSpacePermissions) => {
+  const flatPermissions: SpacePermissions = {}
```
Comment (tusharojha) [inline packages/api/src/subsocial/flatteners/index.ts:67]: Can you add a comment describing it's usage for devs?

File: packages/api/src/subsocial/index.ts [hunk 3]
```diff
@@ -1,215 +1,155 @@
...
+  findPostStruct: (id: AnyId) => Promise<PostStruct | undefined>
+
+  findPost: (query: FindPostQuery) => Promise<PostData | undefined>
+  findPublicPosts: (ids: AnyId[]) => Promise<PostData[]>
+  findPostWithSomeDetails: (query: FindPostQuery) => Promise<PostWithSomeDetails | undefined>
+  findPostWithAllDetails: (id: AnyId) => Promise<PostWithAllDetails | undefined>
+  findPostsWithAllDetails: (query: FindPostsQuery) => Promise<PostWithAllDetails[]>
+  findPublicPostsWithSomeDetails: (query: FindPostsWithDetailsQuery) => Promise<PostWithSomeDetails[]>
+  findPublicPostsWithAllDetails: (ids: AnyId[]) => Promise<PostWithAllDetails[]>
+  findUnlistedPostsWithAllDetails: (ids: AnyId[]) => Promise<PostWithAllDetails[]>
```
Comment (tusharojha) [inline packages/api/src/subsocial/index.ts:37]: In future, it makes sense to create a simple method like findPost with filters to get owner details and all details.

**Review by tusharojha — APPROVED** [review]

File: packages/api/src/subsocial/basic.ts (hunk 1 above)
Comment (olehmell) [reply packages/api/src/subsocial/basic.ts:96]: But it isn't only Owner it is withOwner to with Space

File: packages/api/src/subsocial/flatteners/index.ts (hunk 2 above)
Comment (olehmell) [reply packages/api/src/subsocial/flatteners/index.ts:67]: Sure, I will add docs for it, and ask you improve my text :)

File: packages/api/src/subsocial/index.ts (hunk 3 above)
Comment (olehmell) [reply packages/api/src/subsocial/index.ts:37]: Make sense. Make sense also delete Unlisted/Public for me it isn't SDK part, UI should be decided what do with data

**Pull request closed without merging**